assets/training/model_management/src/azureml/model/mgmt/processors/pyfunc/clip/clip_embeddings_mlflow_wrapper.py

```diff
@@ -0,0 +1,195 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+"""MLflow PythonModel wrapper class that loads the MLflow model, preprocess inputs and performs inference."""
+
+import mlflow
+from PIL import Image
+import pandas as pd
+import torch
+import tempfile
+
+from transformers import AutoProcessor, AutoModelForZeroShotImageClassification
+from config import MLflowSchemaLiterals, MLflowLiterals, Tasks
+from typing import List, Tuple
+
+try:
+    # Use try/except since vision_utils is added as part of model export and not available when initializing
+    # model wrapper for save_model().
+    from vision_utils import create_temp_file, process_image_pandas_series, get_current_device
+except ImportError:
+    pass
+
+
+class CLIPEmbeddingsMLFlowModelWrapper(mlflow.pyfunc.PythonModel):
```

Comment:
Why not extend the existing class? Seems the constructor and `load_context()` method can stay the same and you could simply call a different `predict()` method according to the task.